q2_moshpit/eggnog/_utils.py

```diff
@@ -26,3 +34,104 @@ def _parse_build_diamond_db_params(arg_key, arg_val) -> List[str]:
         return [f"--{arg_key}"]
     else:
         return [f"--{arg_key}", str(arg_val)]
+
+
+def _download_and_build_hmm_db(taxon_id) -> HmmerDirFmt:
+    hmmer_db = HmmerDirFmt()
+    with tempfile.TemporaryDirectory() as tmp:
+        _try_wget(
+            f"{tmp}/{taxon_id}_hmms.tar.gz",
+            "http://eggnog5.embl.de/download/eggnog_5.0/per_tax_level/"
+            f"{taxon_id}/{taxon_id}_hmms.tar.gz",
+            "Error during HMMER database download"
+        )
+
+        # Extracting
+        print(colorify("Decompressing..."))
+        run_command(
+            cmd=["tar", "zxf", f"{taxon_id}_hmms.tar.gz"],
+            cwd=tmp
+        )
+
+        # Merge hmm files + write .hmm.idmap
+        print(colorify("Merging hmm files..."))
+        hmms_merged_p = f"{str(hmmer_db)}/{taxon_id}.hmm"
+        idmap_p = f"{str(hmmer_db)}/{taxon_id}.hmm.idmap"
+
+        # Open output files
+        with open(hmms_merged_p, "a") as hmms, open(idmap_p, "a") as idmap:
+
+            # Iterate through all decompressed files
+            for root, dirnames, files in os.walk(f"{tmp}/{taxon_id}"):
+                for i, file in tqdm(enumerate(files, start=1)):
+                    if file.endswith(".hmm"):
+
+                        # process hmm files
+                        with open(f"{root}/{file}", "r") as hmm_file:
+                            lines = hmm_file.readlines()
+
+                            # Find "NAME" line
+                            for j, line in enumerate(lines):
+                                if line.startswith("NAME "):
+                                    modified_line = re.sub(
+                                        r'\.faa\.final_tree(\.fa)?', "", line
```

Comment:
Sorry this is a bit nit-picky, but these quotes should all be consistent with the rest of the file.

```suggestion
                                        r"\.faa\.final_tree(\.fa)?", "", line
```

Reply:
no worries :)